src/algebra/lie_algebra.lean

```diff
@@ -468,6 +485,15 @@ def lie_ideal_subalgebra (I : lie_ideal R L) : lie_subalgebra R L := {
   lie_mem := by { intros x y hx hy, apply lie_mem_right, exact hy, },
   ..I.to_submodule, }
 
+/-- A Lie module is irreducible if its only non-trivial Lie submodule is itself. -/
+class lie_module.is_irreducible [lie_module R L M] : Prop :=
+(irreducible : ∀ (M' : lie_submodule R L M), (∃ (m : M'), m ≠ 0) → (∀ (m : M), m ∈ M'))
```

Comment:
I think it might be helpful to provide some helper lemmas, like
`is_irreducible_iff_eq_bot_or_top`.
Do you already have a lattice structure on lie submodules?

Reply:
Agreed. We don't have the lattice structure yet; I've been meaning to get round to it so I'll use this as an excuse and do it as part of the changes you suggest (and maybe even change the definition to `irreducible : ∀ (M' : lie_submodule R L M), M' = ⊥ ∨ M' = ⊤`).

Probably worth a lemma saying that the zero module is irreducible too. I'll get on it over the weekend.

Reply:
Ok, if we don't have the lattice structure yet, I'll kick this on the merge queue now. Because it's unrelated to this one, so it's better to put it in a new PR.